libkmod/libkmod-index.c

```diff
@@ -135,6 +135,40 @@ enum node_offset {
 	INDEX_NODE_MASK = 0x0FFFFFFF, /* Offset value */
 };
 
+/* Write buffer to imitate FILE I/O optimization */
+struct wbuf {
+	char bytes[4096];
```

Comment:
Nits:
 - let's drop the comment and use more indicative name (say "write_buffer")
 - could probably bump this to 16K or more based on the stats (aka perf and/or how large our buffers are)

Reply:
Can you elaborate on the stack concern? Should we be using `-Wframe-larger-than=XXX` (as a separate commit/PR)?

Reply:
There is no need to exceed 4 KB (a page size). Using a page size is already very good, after all it's what FILE implementations chose to do as well. Copying 2 or more pages won't gain much, especially in our scenario with only a few hundred write calls left.

I would have more of a concern due to musl-based systems which allocate only 128 KB stacks for threads. Since we are in library code here, let's not exceed what we already do by much more due to PATH_MAX (4096) etc.

Reply:
> Can you elaborate on the stack concern? Should we be using `-Wframe-larger-than=XXX` (as a separate commit/PR)?

That might make sense imho

Reply:
Since we have `struct strbuf sbuf`, I switched from `wbuf` to `struct wrtbuf wbuf`. Looks nicer in functions where `sbuf` and `wbuf` are used next to each other. ;)